### PR TITLE
Inform User that row deletion is not possible if only one row remains in the spreadsheet. 

### PR DESCRIPTION
**Pull request by Steffengreiner**
**What was changed**
The user is now unable to delete the final row(outside of the columnheaders) and is informed via a notification that this is not possible. 

**Technical Note**
I wanted to propagate the notification via an applicationexception within the layout, however this leads to the spreadsheet component breaking entirely, which is why a dedicated notification was necessary.

### CODE REVIEW

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleRegistrationSpreadsheet.java [hunk 1]
```diff
@@ -219,10 +219,6 @@ private void generateRowHeaderCell(int colIndex, int rowIndex) {
    * @param index 0-based row index of the row to remove
    */
   public void deleteRow(int index) {
-    if (getRows() == 1) {
-      // only one row remaining -> the header row
-      return;
-    }
```
Comment (KochTobi) [inline vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleRegistrationSpreadsheet.java:225]: this needs to stay in. The last row can never be deleted (no matter what assumption is taken before).

**Review by KochTobi — CHANGES_REQUESTED** [review]
Hi @Steffengreiner 
Nice work! Two minor things.

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleSpreadsheetLayout.java [hunk 2]
```diff
@@ -76,14 +84,28 @@ private void initButtonLayout() {
...
+        });
     registerButton.addThemeVariants(ButtonVariant.LUMO_PRIMARY);
     sampleInformationButtons.add(backButton, addRowButton, deleteRowButton, cancelButton,
         registerButton);
     add(sampleInformationButtons);
   }
 
+  private void initNotifications() {
+    addComponentAsFirst(lastRowDeletionNotification);
+  }
```
[the remark this answers is not in the document]
Comment (Steffengreiner) [reply vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleSpreadsheetLayout.java:103]: Good point. Do you think it makes sense to throw an exception here?

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleRegistrationSpreadsheet.java (hunk 1 above)
Comment (Steffengreiner) [reply vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleRegistrationSpreadsheet.java:225]: Good point. Do you think it makes sense to throw an exception here?

Comment (KochTobi) [reply vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleRegistrationSpreadsheet.java]: no it is not that exceptional and I do not think anyone wants to delete the header

**Review by KochTobi — APPROVED** [review]

**Pull request merged**